source/Missions/Services/Network/NetworkAgentRegistry.cs

```diff
@@ -80,18 +109,50 @@ public bool RegisterNetworkControlledAgent(NetPeer peer, Guid agentId, Agent age
         {
             if (m_OtherAgents.TryGetValue(peer, out AgentGroupController controller))
             {
-                if (controller.Contains(agent)) return false;
-                if (controller.Contains(agentId)) return false;
+                if (controller.Contains(agent) || controller.Contains(agentId)) return false;
+
+                controller.AddAgent(agentId, agent);
+                m_AgentToId.Add(agent, agentId);
+            }
+            else
+            {
+                AgentGroupController newGroupController = new AgentGroupController();
+                newGroupController.AddAgent(agentId, agent);
+                m_AgentToId.Add(agent, agentId);
+                m_OtherAgents.Add(peer, newGroupController);
+            }
+
+            return true;
+        }
+
+        public bool RegisterPlayerAgent(Guid agentId, Agent agent)
+        {
+            if (m_AgentToId.ContainsKey(agent) ||
```

Comment:
It's slightly better to keep this as 2 if statements just for constancy as well as readability